src/sentry/auth/access.py

```diff
@@ -85,15 +89,27 @@ def has_team(self, team):
     def has_team_access(self, team):
         if not self.is_active:
             return False
+        if self.open_access_policy and self.organization_id == team.id:
+            return True
         return team in self.teams
 
-    def has_team_membership(self, team):
```

Comment:
fwiw we can easily keep has_team_membership, but it seems unused so Id rather axe it unless we have a use-case